anchor/keygen/src/lib.rs

```
@@ -83,11 +76,25 @@ pub fn run_keygen(keygen: Keygen) -> Result<(), KeygenError> {
         KeygenError::Output(format!("Failed to convert output data to JSON string: {e}"))
     })?;
 
-    // Write the JSON file
-    fs::write(&json_file, pretty_json)
-        .map_err(|e| KeygenError::Output(format!("Failed to write keys to JSON file: {e}")))?;
+    if !pem_file.exists() && !json_file.exists() {
+        // Write the PEM file
+        fs::write(&pem_file, &private_pem).map_err(|e| {
+            KeygenError::Output(format!("Failed to write private key to PEM file: {e}"))
+        })?;
 
-    info!("JSON keys written to: {}", json_file.display());
+        info!("Private key written to: {}", pem_file.display());
+
+        // Write the JSON file
+        fs::write(&json_file, pretty_json)
+            .map_err(|e| KeygenError::Output(format!("Failed to write keys to JSON file: {e}")))?;
+
+        info!("JSON keys written to: {}", json_file.display());
+    } else {
+        warn!(
+            "PEM file or JSON file already exist in {:?}, skipping write",
+            output_dir
+        );
```

I think we should return an error here to abort client startup. It makes no sense to operate on a key which has not been saved.

---

Might be relevant https://github.com/sigp/anchor/pull/43#discussion_r1876112496

---

An important difference here is that we need the operator key to perform our duties. If we lose it or overwrite it, we can not act as that operator again, especially as (opposed to validator keys) the operator key is not derived from a mnemonic. So we should be very careful with it.